giocomo_lab_to_nwb/wen22/wen21behaviorinterface.py

```diff
@@ -0,0 +1,138 @@
+                file_epoch_name = file_epoch_name[1:]
+
+            df_data = pd.read_csv(reward_file_path, sep="\t", names=["reward_time_stamps", "x1"])
+
+            df_data_list.append(df_data)
+
+        df_data_concatenated = pd.concat(df_data_list)
+        df_data_concatenated.sort_values(by="reward_time_stamps", inplace=True)
+
+        events = Events(
+            name=f"reward_times",
+            description="timestamps for rewards",
+            timestamps=df_data_concatenated.reward_time_stamps.values,
+        )
+        behavior_module.add(events)
+
+        # Add trial time intervals
+        file_path_list = list(session_path.glob("*trial_times.txt"))
+        file_path_list = [path for path in file_path_list if track_label in path.name]
+        df_data_list = []
+        for trial_file_path in file_path_list:
+            trial_file_name = trial_file_path.name
+            file_epoch_name = trial_file_name.split("_trial")[0].partition("train1")[-1]
+            df_data = pd.read_csv(
+                trial_file_path,
+                sep="\t",
+                names=["stop_time", "x1", "x2", "x3"],
+            )
+            df_data_list.append(df_data)
+
+        df_data_concatenated = pd.concat(df_data_list)
+        df_data_concatenated.sort_values(by="stop_time", inplace=True)
+        df_data_concatenated["start_time"] = df_data_concatenated.stop_time.shift(1).fillna(0)
+        rows_as_dicts = df_data_concatenated[["start_time", "stop_time"]].T.to_dict().values()
+        [nwbfile.add_trial(**row_dict) for row_dict in rows_as_dicts]
+
+        # Add positions
+        file_path_list = list(session_path.glob("*position.txt"))
+        file_path_list = [path for path in file_path_list if track_label in path.name]
+        df_data_list = []
+        for position_file_path in file_path_list:
+            position_file_name = position_file_path.name
+            file_epoch_name = position_file_name.split("_position")[0].partition("train1")
+            if len(file_epoch_name) > 0:
+                file_epoch_name = file_epoch_name[-1]
+
+            df_data = pd.read_csv(
+                position_file_path,
+                sep="\t",
+                names=["position", "timestamps", "x1", "x2"],
+            )
+            df_data["epoch"] = file_epoch_name[1:] if file_epoch_name else no_name_epoch_name
+            df_data_list.append(df_data)
+
+        df_data_concatenated = pd.concat(df_data_list)
+        df_data_concatenated.sort_values(by="timestamps", inplace=True)
+
+        pos_obj = Position(name=f"position during epoch {file_epoch_name[1:]}")
+
+        spatial_series_object = SpatialSeries(
+            name="position",
+            description="position within the virtual reality wheel",
+            data=H5DataIO(df_data_concatenated.position.values, compression="gzip"),
+            reference_frame="unknown",
+            unit="cm",
+            conversion="0.100",
```

Comment:
Can you make the units in meters instead?

Reply:
You mean before passing it to the SpatialSeries so it is already in meters?

Reply:
`SpatialSeries` should be in units of meters, not centimeters. You can use the `conversion` argument instead of multiplying the data

Reply:
Got you.